src/XMakeBuildEngine/Evaluation/LazyItemEvaluator.IncludeOperation.cs

```diff
@@ -61,14 +58,14 @@ protected override ICollection<I> SelectItems(ImmutableList<ItemData>.Builder li
                     }
                 }
 
-                foreach (var operation in _operations)
+                foreach (var fragment in _itemSpec.Fragments)
                 {
-                    if (operation.Item1 == ItemOperationType.Expression)
+                    if (fragment is ItemExpressionFragment<P, I>)
```

Comment:
One funky idea is to also let ItemSpec be responsible for producing string values for each fragment type. This would eliminate this type switching in the include operation. If I find another code clone doing the same thing I'll look into it.